web/src/filter.ts

```diff
@@ -1549,6 +1560,13 @@ export class Filter {
         return terms_mixed_case.map((term: NarrowTerm) => Filter.canonicalize_term(term));
     }
 
+    adjust_with_operand_to_message(msg_id: number): void {
+        const narrow_terms = this._terms.filter((term) => term.operator !== "with");
+        const adjusted_with_term = {operator: "with", operand: `${msg_id}`};
+        const adjusted_terms = [...narrow_terms, adjusted_with_term];
+        this._terms = adjusted_terms;
+    }
```

Comment:
How confident do you feel in the data model of having the `with` clause stored in the `_terms`, rather than, say, as a `_with_message_id` field on `Filter` itself? 

I've not really had a chance to analyze the trade-off, but that seems like one of the key software design questions here.